sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -254,3 +271,197 @@ def _predict_interval(self, fh, X, coverage):
+        clone_self.estimator = estimator.fit(y=y, X=X)
+
+        return clone_self
+
+    def predict(self, h, X=None, level=None):
+        """Make forecasts.
+
+        Parameters
+        ----------
+        h : int
+            Forecast horizon.
+        X : typing.Optional[numpy.ndarray], default=None
+            Optional exogenous of shape (h, n_x).
+        level : typing.Optional[typing.Tuple[int]], default=None
+            Confidence levels (0-100) for prediction intervals.
+
+        Returns
+        -------
+        y_pred : dict
+            Dictionary with entries mean for point predictions and level_* for
+            probabilistic predictions.
+        """
+        mean = self.estimator.predict(fh=range(1, h + 1), X=X)[:, 0]
+        if level is None:
+            return {"mean": mean}
+
+        level = sorted(level)
+        coverage = [round(1 - (_l / 100), 2) for _l in level]
+
+        pred_int = self.estimator.predict_interval(
+            fh=range(1, h + 1), X=X, coverage=coverage
+        )
+
+        return {
+            "mean": mean,
+            **{
+                f"lo-{_l}": pred_int[("Coverage", c, "lower")].values
+                for c, _l in zip(reversed(coverage), reversed(level))
+            },
+            **{
+                f"hi-{_l}": pred_int[("Coverage", c, "upper")].values
+                for c, _l in zip(coverage, level)
+            },
+        }
+
+    def predict_in_sample(self, level=None):
+        """Access fitted MSTL insample predictions.
+
+        Parameters
+        ----------
+        level : typing.Optional[typing.Tuple[int]]
+            Confidence levels (0-100) for prediction intervals.
+
+        Returns
+        -------
+        y_pred : dict
+            Dictionary with entries mean for point predictions and level_* for
+            probabilistic predictions.
+        """
+        if level and not hasattr(self.estimator, "predict_interval"):
+            raise Exception(
+                "The provided trend_forecaster does not support interval prediction."
+                "Please remove the 'level' parameter or use a different model that"
+                "supports it."
+            )
+
+        fitted = self.estimator.predict(fh=self.in_sample_fh, X=self.in_sample_X)[:, 0]
+
+        if level is None:
+            return {"fitted": fitted}
+
+        level = sorted(level)
+        coverage = [round(1 - (_l / 100), 2) for _l in level]
+        pred_int = self.estimator.predict_interval(
+            fh=self.in_sample_fh, X=self.in_sample_X, coverage=coverage
+        )
+
+        return {
+            "fitted": fitted,
+            **{
+                f"fitted-lo-{_l}": pred_int[("Coverage", c, "lower")].values
+                for c, _l in zip(reversed(coverage), reversed(level))
+            },
+            **{
+                f"fitted-hi-{_l}": pred_int[("Coverage", c, "upper")].values
+                for c, _l in zip(coverage, level)
+            },
+        }
```

Comment:
Same comment as above "Coverage" area.

Also, can you create an internal method to reduce duplication?

Reply:
Yes, method for this makes sense